ui/tests/acceptance/volumes-list-test.js

```diff
@@ -1,26 +1,140 @@
+    assert.equal(
+      volumeRow.controllerHealth,
+      `${controllerHealthStr} (${volume.controllersHealthy}/${volume.controllersExpected})`
+    );
+    assert.equal(
+      volumeRow.nodeHealth,
+      `${nodeHealthStr} (${volume.nodesHealthy}/${volume.nodesExpected})`
+    );
+    assert.equal(volumeRow.provider, volume.provider);
+    assert.equal(volumeRow.allocations, readAllocs.length + writeAllocs.length);
+  });
+
+  test('each volume row should link to the corresponding volume', async function(assert) {
+    const volume = server.create('csi-volume');
+
+    await VolumesList.visit();
+
+    await VolumesList.volumes.objectAt(0).clickName();
+    assert.equal(currentURL(), `/csi/volumes/${volume.id}`);
+
+    await VolumesList.visit();
+    assert.equal(currentURL(), '/csi/volumes');
+
+    await VolumesList.volumes.objectAt(0).clickRow();
+    assert.equal(currentURL(), `/csi/volumes/${volume.id}`);
   });
 
-  skip('visiting /csi/volumes', async function() {});
+  test('when there are no volumes, there is an empty message', async function(assert) {
+    await VolumesList.visit();
 
-  skip('/csi/volumes should list the first page of volumes sorted by name', async function() {});
+    assert.ok(VolumesList.isEmpty);
+    assert.equal(VolumesList.emptyState.headline, 'No Volumes');
+  });
+
+  test('when the namespace query param is set, only matching volumes are shown and the namespace value is forwarded to app state', async function(assert) {
+    server.createList('namespace', 2);
+    const volume1 = server.create('csi-volume', { namespaceId: server.db.namespaces[0].id });
+    const volume2 = server.create('csi-volume', { namespaceId: server.db.namespaces[1].id });
+
+    await VolumesList.visit();
+
+    assert.equal(VolumesList.volumes.length, 1);
+    assert.equal(VolumesList.volumes.objectAt(0).name, volume1.id);
 
-  skip('each volume row should contain information about the volume', async function() {});
+    const secondNamespace = server.db.namespaces[1];
+    await VolumesList.visit({ namespace: secondNamespace.id });
 
-  skip('each volume row should link to the corresponding volume', async function() {});
+    assert.equal(VolumesList.volumes.length, 1);
+    assert.equal(VolumesList.volumes.objectAt(0).name, volume2.id);
+  });
+
+  test('when accessing volumes is forbidden, a message is shown with a link to the tokens page', async function(assert) {
+    server.pretender.get('/v1/volumes', () => [403, {}, null]);
 
-  skip('when there are no volumes, there is an empty message', async function() {});
+    await VolumesList.visit();
+    assert.equal(VolumesList.error.title, 'Not Authorized');
 
-  skip('when the namespace query param is set, only matching volumes are shown and the namespace value is forwarded to app state', async function() {});
+    await VolumesList.error.seekHelp();
+    assert.equal(currentURL(), '/settings/tokens');
+  });
```

Comment:
Nice to see this, I feel like I should have done something similar in a couple of places 😳